File: python/104_easy.py
class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        queue = collections.deque([(root, 1)])
        result = 0
        while queue:
            node, count = queue.popleft()

            if node:
                result = max(result, count)
                queue.append((node.left, count+1))
                queue.append((node.right, count+1))

File: java/104_easy.java
class Solution {
    public int maxDepth(TreeNode root) {
        if (root == null) return 0;

        int left = maxDepth(root.left);
        int right = maxDepth(root.right);
        int height;
        if (left > right) height = 1 + left;
        else height = 1 + right;

        return height;
    }
}